cpu/cortexm_common/mpu.c

```diff
@@ -59,11 +57,17 @@ bool mpu_enabled(void) {
 }
 
 int mpu_configure(uint_fast8_t region, uintptr_t base, uint_fast32_t attr) {
-/* Todo enable MPU support for Cortex-M23/M33 */
-#if __MPU_PRESENT && !defined(CPU_CORE_CORTEX_M23)
+#if __MPU_PRESENT
+
+#if __ARM_ARCH_7M__ || __ARM_ARCH_7EM__
     MPU->RNR  = region;
     MPU->RBAR = base & MPU_RBAR_ADDR_Msk;
     MPU->RASR = attr | MPU_RASR_ENABLE_Msk;
+#elif __ARM_ARCH_8M_MAIN__ || __ARM_ARCH_8M_BASE__
+    MPU->RNR = region;
+    MPU->RBAR = base;
+    MPU->RLAR = attr;
+#endif
```

Comment:
Did you try this?
Would be great if this fixes the MPU on `saml1x` (Cortex-M23) as well.

I should be able to test this tomorrow.

Reply:
Well we don't provide the `cortexm_mpu` feature on `saml1x` anyway, so this won't break anything.

Reply:
I don't think this worked. The attributes have changed, and they are structured using some #define that looks like it was Cortex-M3/M4 only.

Reply:
[This](https://github.com/RIOT-OS/RIOT/blob/77735e7e10da53b6e1b534c6999e7252a99f8c9b/core/sched.c#L208) I mean.

Reply:
Murdock is [not happy](https://ci.riot-os.org/RIOT-OS/RIOT/14367/1d29eec70186db1c9b67b2a7a3b41c51440c48be/output/compile/examples/riot_and_cpp/nucleo-g070rb:gnu.txt)